Answers.md

```diff
@@ -0,0 +1,57 @@
+### Chapter 10 Self Test Answers
+
+1) Why does Java define both byte and character streams.
+	
+	So that the streams can be optimised for the type of data being output. 
+
+2)
+	
+	Byte streams were all that was available at the time System.in and System.out were originally defined.
+
+3)
+
+	FileInputStream in = new FileInputStream("inFile.bin");
+
+4)
+
+	FileReader in = new FileReader("inFile.txt");
+
+5)	
+
+	RandomAcessFile in = new RandomAcessFile("inFile.bin", "rw");
+
+6) 
+
+	double d = Double.parse("123.23");
```

Comment:
For whatever reason, they opted for `T.parseT(String)` for the primitive wrapper types, though `T.parse` makes more sense to me. This would need to be `Double.parseDouble`.